bundles/org.openhab.binding.insteon/src/main/java/org/openhab/binding/insteon/internal/handler/InsteonNetworkHandler.java

```diff
@@ -104,9 +106,24 @@ public void initialize() {
                             insteonBinding.logDeviceStatistics();
                         }, 0, LOG_DEVICE_STATISTICS_DELAY_IN_SECONDS, TimeUnit.SECONDS);
 
```

Comment:
I checked the code of the `initialize` method with all these nested asynchronous calls. For me, the higher `scheduler.execute` should be simply removed, it will allow `initialze` setting a thing status before returning, which is what is expected by the framework.

Reply:
OK, I removed it, but it was originally done this way because Kai asked me to do it that way.